### ISSUE TITLE
Cursor mode.

### ISSUE THREAD

**Issue by kk6** [kk6/python-annict#11]
For example:

```python

class API:

    def works(self, ...):
        ...
    works.cursor_support = True
    
    def me(self, ...):
        ...
    
    def set_status(self, ...):
        ...

class SimpleCursor(object):

    def __init__(self, method, **kwargs):
        if not hasattr(method, 'cursor_support'):
            raise TypeError(f"Cursor does not support this method: {method.__func__.__qualname__}")
        self.method = method
        self.kwargs = kwargs
        if 'page' not in self.kwargs:
            self.kwargs['page'] = 1

    def cursor(self):
        while 1:
            results = self.method(**self.kwargs)
            for result in results:
                yield result
            self.kwargs['page'] += 1
            if not results.next_page or not results:
                raise StopIteration

```

Usage:

```python

>>> for user in annict.utils.SimpleCursor(api.search_users, per_page=50, sort_id='desc').cursor():
...     print(user)
...

<User object>
<User object>


>>> for user in annict.utils.SimpleCursor(api.me).cursor():
...     print(user)
...
  
Traceback (most recent call last):
  ...
TypeError: This method does not correspond to Cursor: API.me


```

**Comment by kk6** [comment]
Closed by #21

**Comment by kk6** [comment]
Closed by #21
